### PR TITLE
[DEV-2378] amlCheck mail block

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations (on [DFX services](https://dev.services.dfx.swiss))
  - Login/logout
  - Buy/sell payment request
  - KYC page

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
Add migration

File: src/subdomains/core/aml/aml.service.ts [hunk 1]
```diff
@@ -46,6 +49,10 @@ export class AmlService {
     if (!entity.userData.lastNameCheckDate) errors.push(AmlError.NO_NAME_CHECK);
     if (Util.daysDiff(entity.userData.lastNameCheckDate) > Config.amlCheckLastNameCheckValidity)
       errors.push(AmlError.OUTDATED_NAME_CHECK);
+    if (
+      blacklist.some((b) => b.type === SpecialExternalAccountType.BANNED_MAIL && entity.userData.mail.includes(b.value))
+    )
+      errors.push(AmlError.SUSPICIOUS_MAIL);
```
Comment (davidleomay) [inline src/subdomains/core/aml/aml.service.ts:55]: Maybe we should use Regex here? Then we can remove the `hasSuspiciousMail` check in the user-data.

Comment (davidleomay) [reply src/subdomains/core/aml/aml.service.ts:55]: Mail might be undefined?

File: src/subdomains/core/aml/aml.service.ts [hunk 2]
```diff
@@ -103,43 +112,64 @@ export class AmlService {
 
       if (entity.bankTx) {
         // bank
-        if (blacklist.some((b) => b.bic && b.bic === entity.bankTx.bic)) errors.push(AmlError.BIC_BLACKLISTED);
-        if (blacklist.some((b) => b.iban && b.iban === entity.bankTx.iban)) errors.push(AmlError.IBAN_BLACKLISTED);
+        if (
+          blacklist.some(
+            (b) => b.type === SpecialExternalAccountType.BANNED_BIC && new RegExp(b.value).test(entity.bankTx.bic),
+          )
```
Comment (davidleomay) [inline src/subdomains/core/aml/aml.service.ts:118]: Maybe we should create a helper method in the SpecialExternalAccount entity?
```ts
matches(type: SpecialExternalAccountType, value: string): boolean {
  return this.type === type && new RegExp(this.value).test(value);
}
```

**Review by davidleomay — APPROVED** [review]

**Pull request merged**